app/assets/javascripts/discourse/components/d-editor.js.es6

```diff
@@ -659,27 +660,9 @@ export default Ember.Component.extend({
     }
 
     if (this.siteSettings.enable_rich_text_paste && html && !handled) {
-      const placeholder = `${ plainText || I18n.t('pasting') }`;
-      const self = this;
-
-      this.appEvents.trigger('composer:insert-text', placeholder);
+      const markdown = toMarkdown(html);
```

Comment:
this is missing stuff imo, the method should always accept raw as well, because it needs to make some decisions based off that. 

For example: 

If the HTML converted stuff is somehow 10 chars long and raw is 1000 ... would you consider using the HTML? This interface offers no way to heuristically decide if we even want to do html conversion or not.

Reply:
additionally can you make sure that IMG paste if it exists in binary in the clipboard always take precedence over html?

Reply:
> skip one of them, the image one. We **should not prioritize image pastes unless there is no
other way** to do this.

but in other [PR](https://github.com/discourse/discourse/pull/5378#issuecomment-347643515) @coding-horror told me that image paste should have low priority.

Reply:
@vinothkannans  see our chat, prioritize single image pastes please not hotlinking a single image.